src/os/bluestore/BlueStore.cc

```diff
@@ -565,6 +566,31 @@ void BlueStore::Cache::_audit_lru(const char *when)
 }
 #endif
 
+struct Int64ArrayMergeOperator : public KeyValueDB::MergeOperator {
+  virtual void merge_nonexistant(
+    const char *rdata, size_t rlen, std::string *new_value) override {
+    *new_value = std::string(rdata, rlen);
+  }
+  virtual void merge(
+    const char *ldata, size_t llen,
+    const char *rdata, size_t rlen,
+    std::string *new_value) {
+    assert(llen == rlen);
+    assert((rlen % 8) == 0);
+    new_value->resize(rlen);
+    __le64* lv = (__le64*)ldata;
+    __le64* rv = (__le64*)rdata;
+    for (size_t i = 0; i < rlen >> 3; ++i) {
+      (__le64&)(new_value->at(i << 3)) = lv[i] + rv[i];
```

Comment:
IMHO ldata is supposed to be an input (read only) parameter. And result to be stored to new_value. Am I missing something?